src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -31,93 +96,1203 @@ const ActorInit En_Go_InitVars = {
+        return true;
+    } else if (DECR(this->unk_21C)) {
+        if ((this->unk_21C & 1)) {
+            this->actor.posRot.pos.y += 1.5f;
+        } else {
+            this->actor.posRot.pos.y -= 1.5f;
+        }
+        return true;
+    } else {
+        this->unk_21A--;
+        if (this->unk_21A <= 0) {
+            if (this->unk_21A == 0) {
+                this->unk_21C = Math_Rand_S16Offset(60, 30);
+                this->unk_21A = 0;
+                this->actor.velocity.y = 0.0f;
+                return true;
+            }
+            this->unk_21A = unk_arg1;
+        }
+        this->actor.velocity.y = ((f32)this->unk_21A / (f32)unk_arg1) * unk_arg2;
+        return true;
+    }
+}
+
+void func_80A3F908(EnGo* this, GlobalContext* globalCtx) {
+    Player* player;
+    f32 float1;
+    s32 isUnkCondition;
+
+    player = PLAYER;
+
+    if (this->actionFunc == EnGo_BiggoronActionFunc || this->actionFunc == EnGo_GoronLinkRolling ||
+        this->actionFunc == EnGo_FireGenericActionFunc || this->actionFunc == EnGo_Eyedrops ||
+        this->actionFunc == func_80A40DCC || this->actionFunc == EnGo_GetItem || this->actionFunc == func_80A40C78 ||
+        this->actionFunc == func_80A40B1C) {
+
+        float1 = (this->collider.dim.radius + 30.0f);
+        float1 *= (this->actor.scale.x / 0.01f);
+
+        if ((this->actor.params & 0xF0) == 0x90) {
+            float1 *= 4.8f;
+        }
+
+        if ((this->actor.params & 0xF0) == 0x90) {
+            isUnkCondition =
+                func_80A3ED24(globalCtx, &this->actor, &this->unk_1E0, float1, EnGo_GetTextID, EnGo_SetFlagsGetStates);
+        } else {
+            isUnkCondition =
+                func_800343CC(globalCtx, &this->actor, &this->unk_1E0, float1, EnGo_GetTextID, EnGo_SetFlagsGetStates);
+        }
+
+        if (((this->actor.params & 0xF0) == 0x90) && (isUnkCondition == true)) {
+
+            if (INV_CONTENT(ITEM_POCKET_EGG) == ITEM_SWORD_BROKEN) {
+                if (func_8002F368(globalCtx) == EXCH_ITEM_SWORD_BROKEN) {
+                    if (gSaveContext.infTable[11] & 0x10) {
+                        this->actor.textId = 0x3055;
+                    } else {
+                        this->actor.textId = 0x3054;
+                    }
+                } else {
+                    this->actor.textId = 0x3053;
+                }
+                player->actor.textId = this->actor.textId;
+            }
+
+            if (INV_CONTENT(ITEM_POCKET_EGG) == ITEM_EYEDROPS) {
+                if (func_8002F368(globalCtx) == EXCH_ITEM_EYEDROPS) {
+                    this->actor.textId = 0x3059;
+                } else {
+                    this->actor.textId = 0x3058;
+                }
+                player->actor.textId = this->actor.textId;
+            }
+        }
+    }
+}
+
+void EnGo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGo* this = THIS;
+    s32 pad;
+    Vec3f D_80A41BD_80A41B9C9C = { 0.0f, 0.0f, 0.0f }; // unused
```

Comment:
what happened to this variable name?

Reply:
Good question. Probably copy-pasta shenanigans